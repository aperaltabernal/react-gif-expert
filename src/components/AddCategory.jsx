import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (evt) => {
        evt.preventDefault();
        if(inputValue.trim().length <= 1) return;
        onNewCategory(inputValue);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar GIF's"
                value={inputValue}
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}