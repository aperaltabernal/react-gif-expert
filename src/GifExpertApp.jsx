import {useState} from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([]);
    
    const onAddCategory = (nuevaCategoria) => {
        if(categories.includes(nuevaCategoria)) return;
        setCategories([nuevaCategoria, ...categories]);
        //setCategories(c => [...c, 'Nuevo']);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={onAddCategory}/>

            <ol>
                {
                    categories.map((category) => 
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </>
        
    )
}
