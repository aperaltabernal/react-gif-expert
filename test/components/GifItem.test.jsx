import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Pruebas en <GifItem />', () => {

    const title = 'CS Lewis';
    const url = 'https://';
    
    test('debe hacer match con el snapshot', () => {

        const { container }  = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={title} url={url} />);
        //  screen.debug();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe mostrar el titulo', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});