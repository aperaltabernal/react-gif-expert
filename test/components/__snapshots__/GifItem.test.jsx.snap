// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="CS Lewis"
      src="https://"
    />
    <p>
      CS Lewis
    </p>
  </div>
</div>
`;
